apps-rendering/src/client/newsletterEmbeds.ts

```diff
@@ -0,0 +1,61 @@
+type NewsletterHeightEventType = { source: MessageEventSource; origin: string };
+type MessageType = {
+	type: string;
+	value: string;
+};
+
+const allowedOrigins = ['https://www.theguardian.com'];
+export default (): Promise<void> => {
+	const allIframes: HTMLIFrameElement[] = [].slice.call(
+		document.querySelectorAll('.email-sub__iframe'),
+	);
+
+	// Tell the iframes to resize once this script is loaded
+	// Otherwise, earlier resize events might be missed
+	// So we don't have to load this script as a priority on each load
+	allIframes.forEach((iframe) => {
+		iframe.contentWindow?.postMessage('resize', '*');
+	});
+
+	window.addEventListener('message', (event) => {
+		if (
+			!allowedOrigins.includes(
+				(event as NewsletterHeightEventType).origin,
```

Comment:
In the scenario where you call `window.addEventListener` with `message` as the first argument, is TypeScript automatically able to narrow the type of `event` to `MessageEvent`? If so, does that interface already have `origin` and `source` defined as properties? In which case are the typecasts here and on line 32 needed?

Reply:
Yes, typescript automatically narrows `event` to `MessageEvent`, so we can remove the typecasts & the `NewsletterHeightEventType`!